controller/src/main/java/io/pravega/controller/store/stream/AbstractStreamMetadataStore.java

```diff
@@ -125,9 +127,17 @@ public OperationContext createContext(String scope, String name) {
                                                    final Executor executor) {
         return getSafeStartingSegmentNumberFor(scope, name)
                 .thenCompose(startingSegmentNumber ->
-                    withCompletion(getStream(scope, name, context).create(configuration, createTimestamp, startingSegmentNumber), executor));
+                    withCompletion(checkScopeExists(scope)
```

Comment:
It need not be atomic. If scope gets deleted while we have not begun to create the stream, subsequent steps may fail and thats ok. 

CheckScopeExists is added here because of the expectation of specific error response in case scope doesnt exist. 

With Zookeeper, having scopes in the same hierarchy of znode meant create would fail if scope did not exist. 
However, with KVS, scopes is a separate table so we need to check there first.